pulp_smash/tests/rpm/api_v2/test_repo_create.py

```diff
@@ -0,0 +1,662 @@
+    poll_limit = 10
+    poll_counter = 0
+    while True:
+        response = requests.get(
+            server_config.base_url + href,
+            **server_config.get_requests_kwargs()
+        )
+        response.raise_for_status()
+        attrs = response.json()
+        if attrs['state'] in _TASK_END_STATES:
+            yield attrs
+            for spawned_task in attrs['spawned_tasks']:
+                yield _poll_tasks(server_config, spawned_task['_href'])
+            break
+        poll_counter += 1
+        if poll_counter > poll_limit:
+            raise _TaskTimedOutException(
+                'Task {} is ongoing after {} polls.'.format(href, poll_limit)
+            )
+        sleep(5)
+
+
+class CreateTestCase(TestCase):
+    """Create two RPM repositories, with and without feed URLs respectively."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two RPM repositories, with and without feeds."""
+        cls.cfg = get_config()
+        cls.bodies = tuple((_gen_rpm_repo_body() for _ in range(2)))
+        cls.bodies[1]['importer_config'] = {'feed': rand_str()}  # invalid feed
+        cls.attrs_iter = tuple((
+            _create_repo(cls.cfg, body) for body in cls.bodies
+        ))
+        cls.importers_iter = tuple((
+            _get_importers(cls.cfg, attrs['_href']) for attrs in cls.attrs_iter
+        ))
+
+    def test_id_notes(self):
+        """Valdiate the ``id`` and ``notes`` attributes for each repo."""
+        for key in ('id', 'notes'):
+            for body, attrs in zip(self.bodies, self.attrs_iter):
+                with self.subTest((key, body, attrs)):
+                    self.assertIn(key, attrs)
+                    self.assertEqual(body[key], attrs[key])
+
+    def test_number_importers(self):
+        """Each repository should have only one importer."""
+        for i, importers in enumerate(self.importers_iter):
+            with self.subTest(i=i):
+                self.assertEqual(len(importers), 1, importers)
+
+    def test_importer_type_id(self):
+        """Validate the ``importer_type_id`` attribute of each importer."""
+        key = 'importer_type_id'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body[key], importers[0][key])
+
+    def test_importer_config(self):
+        """Validate the ``config`` attribute of each importer."""
+        key = 'config'
+        for body, importers in zip(self.bodies, self.importers_iter):
+            with self.subTest((body, importers)):
+                self.assertIn(key, importers[0])
+                self.assertEqual(body['importer_' + key], importers[0][key])
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroy the created RPM repositories."""
+        for attrs in cls.attrs_iter:
+            _delete(cls.cfg, attrs['_href'])
+
+
+class SyncFeedTestCase(TestCase):
```

Comment:
The `SyncFeedTestCase` and `SyncInvalidFeedTestCase` have basically identical set-up procedures. I might as well just combine the two classes. :+1: